lib/opFns.js

```diff
@@ -157,6 +157,45 @@ module.exports = {
 
     return new BN(word.shrn((31 - pos.toNumber()) * 8).andln(0xff))
   },
+  SHL: function (a, b, runState) {
+    if (!runState._common.gteHardfork('constantinople')) {
+      trap(ERROR.INVALID_OPCODE)
+    }
+    if (a.gten(256)) {
+      return new BN(0)
+    }
+    return b.shln(a.toNumber()).iand(utils.MAX_INTEGER)
+  },
+  SHR: function (a, b, runState) {
+    if (!runState._common.gteHardfork('constantinople')) {
+      trap(ERROR.INVALID_OPCODE)
+    }
+    if (a.gten(256)) {
+      return new BN(0)
+    }
+    return b.shrn(a.toNumber())
```

Comment:
https://github.com/ethereum/EIPs/blob/master/EIPS/eip-145.md#0x1c-shr-logical-shift-right

Same for ``SHR``.